### PR TITLE
feat(users): Add profile level invites

### LINKED ISSUE #5792: feat(users): Add profile level invites
Currently invites are at merchant level only. Invite, accept invite and accept invite from email should change because of this.

### PR DESCRIPTION
**Pull request by ThisIsMani**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

- Removed preferred_merchant_id from update user API.
- Invites will work at profile level as well.
- Handle internal users in list APIs
  - Org List for internal users will throw error.
  - Merchant list will list merchants.
  - Profile list will list profiles.
- New API for accept invite and merchant select is created which takes entity as request.
- Email token will now have a entity instead of merchant_id.

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Closes #5792 

## How did you test it?

All the following APIs should work without any issues.
- Invite multiple
- Accept invite from email
- Accept invite
- Merchant select

- List orgs for user will throw the following error
```
{
    "error": {
        "type": "invalid_request",
        "message": "User Role Operation Not Supported",
        "code": "UR_23"
    }
}
```

New APIs
Accept Invite (works with JWT Token)
```
curl --location 'http://localhost:8080/user/user/invite/accept/v2' \
--header 'Content-Type: application/json' \
--header 'Authorization: ••••••' \
--data '[
    {
        "entity_id": "merchant_id",
        "entity_type": "merchant"
    }
]'
```
Response: 200 OK

Accept Invite (works with Merchant Select SPT)
```
curl --location 'http://localhost:8080/user/user/invite/accept/v2/pre_auth' \
--header 'Content-Type: application/json' \
--header 'Authorization: ••••••' \
--data '[
    {
        "entity_id": "merchant_id",
        "entity_type": "merchant"
    }
]'
```
Response: 
```json
{
    "token": "token",
    "token_type": "user_info"
}
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/types/domain/user.rs [hunk 1]
```diff
@@ -1421,3 +1393,16 @@ impl NewUserRole<ProfileLevel> {
             .ok_or(report!(UserErrors::InternalServerError))
     }
 }
+//
```
Comment (racnan) [inline crates/router/src/types/domain/user.rs:1396]: ?

File: crates/router/src/core/user.rs [hunk 2]
```diff
@@ -2333,63 +2390,59 @@ pub async fn list_merchants_for_user_in_org(
     )
     .await
     .change_context(UserErrors::InternalServerError)?;
-    let merchant_accounts = if role_info.get_entity_type() == EntityType::Organization {
-        state
+    let merchant_accounts = match role_info.get_entity_type() {
+        EntityType::Organization | EntityType::Merchant | EntityType::Internal => state
```
Comment (racnan) [inline crates/router/src/core/user.rs:2394]: `Merchant` shouldn't be here.

**Review by racnan — COMMENTED** [review]
above comments

File: crates/router/src/core/user.rs [hunk 3]
```diff
@@ -803,40 +864,61 @@ pub async fn resend_invite(
...
         user_name: domain::UserName::new(user.get_name())?,
         settings: state.conf.clone(),
         subject: "You have been invited to join Hyperswitch Community!",
-        merchant_id: user_from_token.merchant_id,
-        auth_id,
+        entity: email_types::Entity {
+            entity_id,
+            entity_type,
+        },
+        auth_id: auth_id.clone(),
```
[the remark this answers is not in the document]
Comment (ThisIsMani) [reply crates/router/src/core/user.rs:942]: We can't.

**Review by lsampras — APPROVED** [review]

**Review by racnan — APPROVED** [review]

**Pull request merged**